File: P10/P41412_en/main.java
import java.util.Scanner;
import java.text.DecimalFormat;
import java.math.RoundingMode;

class Main {

  public static void main(String[] args) {
    final Scanner sc = new Scanner(System.in);
    final DecimalFormat df = new DecimalFormat("0.0000");
    df.setRoundingMode(RoundingMode.HALF_EVEN);
    while (sc.hasNextInt()) {
      int n = sc.nextInt();
      double[] v = new double[n];
      for (int i = 0; i < n; ++i) v[i] = sc.nextDouble();
      Solution.insertionSort(v);
      for (int i = 0; i < n; ++i) System.out.print(" " + df.format(v[i]));
      System.out.println();
    }
  }

}

File: P10/P41412_en/main.c
#include <stdio.h>

void insertion_sort(int n, double v[n]);

int main()
{
    int n, i;
    while (scanf("%d", &n) > 0) {
        double v[n];
        for (i = 0; i < n; ++i)
            scanf("%lf", &v[i]);
        insertion_sort(n, v);
        for (i = 0; i < n; ++i)
            printf(" %.4lf", v[i]);
        printf("\n");
    }
    return 0;
}
